modules/ghlint-api/src/main/kotlin/net/twisterrob/ghlint/model/Permissions.kt

```diff
@@ -0,0 +1,43 @@
+package net.twisterrob.ghlint.model
+
+// https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
+
+@Suppress("detekt.ComplexInterface")
+public interface Permissions {
+	public val actions: Access
+	public val attestations: Access
+	public val checks: Access
+	public val contents: Access
+	public val deployments: Access
+	public val idToken: Access
+	public val issues: Access
+	public val metadata: Access
+	public val packages: Access
+	public val pages: Access
+	public val pullRequests: Access
+	public val repositoryProjects: Access
+	public val securityEvents: Access
+	public val statuses: Access
+
+	public val map: Map<String, String>
+
+	@Suppress("detekt.CyclomaticComplexMethod")
+	public fun accessOf(permission: Permission) : Access {
```

Comment:
unit tests? I would do a `val permissions: Permissions = mock()` then stub one val, and use accessOf to get the stubbed value. Sadly not parameterizable, but you have just the tool :D